src/effects.js

```diff
@@ -631,15 +633,25 @@ jQuery.each( {
 } );
 
 jQuery.timers = [];
+var oldLength = 0;
 jQuery.fx.tick = function() {
 	var timer,
 		i = 0,
 		timers = jQuery.timers;
 
 	fxNow = jQuery.now();
+	oldLength = timers.length;
+	for ( ; i < oldLength; i++ ) {
+		var removedTimers = 0;
 
-	for ( ; i < timers.length; i++ ) {
-		timer = timers[ i ];
+		for ( var j = 0; j < lis.length; j++ ) {
```

Comment:
I'm hoping to avoid O(N²) operations and extra global variables, and was thinking we'd do this by checking/clearing `timer.elem` since that's already cleared in an `always` callback.